fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -74,13 +67,37 @@ internal class HttpClient(
         return retrieveResponse(request, connection)
     }
 
+    /**
+     * When method is Method.PATCH, use reflection to set private field "method" on the HttpURLConnection instance
+     * For other methods, fall back to the public setter "setRequestMethod"
+     */
+    private fun HttpURLConnection.forceMethod(method: Method) {
+        when (method) {
+            Method.PATCH -> {
+                try {
+                    (this.javaClass.getDeclaredField("delegate").apply { this.isAccessible = true }.get(this) as HttpURLConnection?)?.forceMethod(method)
+                } catch (ex: NoSuchFieldException) {
+                    // ignore
```

Comment:
Okay but _can_ it fail? Is this field guaranteed to be as it is on _all_ JVM implementatons of `HttpURLConnection`, if the url is yes, this comment should say that. If the answer is no, the fallback should be the header override.